src/test/java/guitests/FilterPersonsGuiTest.java

```diff
@@ -0,0 +1,119 @@
+package guitests;
+
+import address.model.datatypes.AddressBook;
+import address.model.datatypes.person.Person;
+import address.model.datatypes.tag.Tag;
+import org.junit.Test;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Collections;
+
+import static org.junit.Assert.assertTrue;
+
+public class FilterPersonsGuiTest extends GuiTestBase {
+    private Person john, mary, annabelle, barney, charlie, danny;
+    private Tag colleagues, friends, family;
+
+    @Override
+    public AddressBook getInitialData() {
+        john = new Person("John", "Tan", 1);
+        mary = new Person("Mary", "Jones", 2);
+        annabelle = new Person("Annabelle", "Muellers", 3);
+        barney = new Person("Barney", "Rogers", 4);
+        charlie = new Person("Charlie", "Chip", 5);
+        danny = new Person("Danny", "Rogers", 6);
+        colleagues = new Tag("colleagues");
+        friends = new Tag("friends");
+        family = new Tag("family");
```

Comment:
Think there will be alot of conflicts if I change this test data though. Will probably need to wait till the other PR is done.